### ISSUE TITLE
[bug] zaya_engine.cpp and backend_hip.cpp both hardcode Zaya1-8B dimensions (H=2048, NQ=8, NKV=2, L=40) — any other model produces silent garbage

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#249]
## Summary

Both `src/zaya_engine.cpp` and `tests/backends/backend_hip.cpp` hardcode the model dimensions for **a single specific model** (Zaya1-8B: H=2048, NQ=8, NKV=2, HD=128, L=40, V=262272) as compile-time constants used in GPU kernel launches. The `backend_hip.cpp` does have a dimension validation guard that refuses to load non-matching models, but `zaya_engine.cpp` has **no such guard** — it silently uses the hardcoded constants regardless of what model config is loaded, producing wrong (but not obviously wrong) output.

## Evidence

**`src/zaya_engine.cpp:19-20`**:
```cpp
static constexpr int H=2048,NQ=8,NKV=2,HD=128,QD=NQ*HD,KD=NKV*HD,QKV=QD+KD;
static constexpr int N_LAYERS=40,VOCAB=262272,N_EXP=16,N_EXP_T=17,N_FF=2048,RTR_H=256;
```
These are used **directly as kernel launch bounds** — e.g. `moe_tiled_gemv<<<QD/16, 128>>>(...)` would use `8*128/16 = 64` grid blocks even if the loaded model has different QD. The runtime config struct `eng` is populated from these same hardcoded values via `#define`/`#undef` tricks and appears unused by the kernel launches.

**`tests/backends/backend_hip.cpp`** has a dimension check at line ~138:
```cpp
if (cfg.hidden_size != 2048 || cfg.num_heads != 8 || ... ) {
    fprintf(stderr, "HIP: refusing to load — would produce silent garbage.\n");
    return false;
}
```
But `zaya_engine.cpp` has no equivalent guard.

**`zaya_engine.cpp` is linked into `unified_server` and `zaya_agent`** — the "one binary, all models" targets. If a user passes a model with e.g. H=4096, the engine will:
1. Allocate buffers for H=2048
2. Launch kernels with wrong grid dimensions
3. Read/write past allocated buffer boundaries
4. Produce NaN logits (or crash on hipMalloc errors from reading the real model's bigger config)

## Impact

- The "all models" claim is immediately broken for any model except Zaya1-8B.
- Unlike `backend_hip.cpp`, there's no "refusing to load" message — the server starts, reports success, and produces garbage silently.

## Suggested

- Add the same dimension validation `backend_hip.cpp` has to `zaya_engine.cpp`, or
- Make the kernels use runtime grid dimensions from the loaded model config instead of compile-time constants, or
- At minimum, print a prominent WARNING when the loaded model dimensions don't match the compiled-in constants.
